drivers/ethernet/eth_mcux.c

```diff
@@ -284,8 +284,6 @@ static void eth_mcux_phy_event(struct eth_context *context)
 	switch (context->phy_state) {
 	case eth_mcux_phy_state_initial:
 #ifdef CONFIG_SOC_SERIES_IMX_RT
-		ENET_StartSMIRead(ENET, phy_addr, PHY_CONTROL2_REG,
```

Comment:
The point of this was to do something like this:

```
        uint32_t data = 0;
        result = PHY_Read(base, phyAddr, PHY_CONTROL2_REG, &data);
        if ( result != kStatus_Success)
        {
            return result;
        }
        result = PHY_Write(base, phyAddr, PHY_CONTROL2_REG, (data | PHY_CTL2_REFCLK_SELECT_MASK));
        if (result != kStatus_Success)
        {
            return result;
        }
```
(Taken from an SDK example)
The reset values of PHY_CONTROL2_REG has jabber counter set (bit 8), the rest is 0.
Writing RMII clock bit to use 50MHz will set all other registers to 0, including bit 8, which has no effect at this moment I assume.

Function `ENET_ReadSMIData` should be used to read the value of the register after initiating `ENET_StartSMIRead `before writing to PHY_CONTROL2_REG.

What do you think?

Reply:
It make sense to do a read/modify/write instead of a "blind" write to this register, but I would avoid using PHY_Read in this function since it uses a busy loop and would block the system work queue.

I'll come up with a proposal tomorrow.

Reply:
> It make sense to do a read/modify/write instead of a "blind" write to this register, but I would avoid using PHY_Read in this function since it uses a busy loop and would block the system work queue.
> 
> I'll come up with a proposal tomorrow.

@agansari I was probably too picky about blocking the work queue since the busy polling doesn't take that long (I measured 50 us). I pushed an additional commit that implements the read/modifying/write as you suggested.